examples/generative/pixelcnn.py

```diff
@@ -0,0 +1,168 @@
+# Concatenate all of the images together
+data = np.concatenate((x, y), axis=0)
+# round all pixel values less than 33% of the max 256 value to 0
+# anything above this value gets rounded up to 1 so that all values are either
+# 0 or 1
+data = np.where(data < (0.33 * 256), 0, 1)
+data = data.astype(np.float32)
+
+"""
+##Create two classes for the requisite Layers for the model
+"""
+
+# the first layer to create will be the PixelCNN layer, this layer simply
+# builds on the 2D convolutional layer but with the requisite masking included
+class PixelConvLayer(layers.Layer):
+    def __init__(self, mask_type, **kwargs):
+        super(PixelConvLayer, self).__init__()
+        self.mask_type = mask_type
+        self.conv = layers.Conv2D(**kwargs)
+
+    def build(self, input_shape):
+        # build the conv2d layer to initialize kernel variables
+        self.conv.build(input_shape)
+        # use said initialized kernel to develop the mask
+        kernel_shape = self.conv.kernel.get_shape()
+        self.mask = np.zeros(shape=kernel_shape)
+        self.mask[: kernel_shape[0] // 2, ...] = 1.0
+        self.mask[kernel_shape[0] // 2, : kernel_shape[1] // 2, ...] = 1.0
+        if self.mask_type == "B":
+            self.mask[kernel_shape[0] // 2, kernel_shape[1] // 2, ...] = 1.0
+
+    def call(self, inputs):
+        self.conv.kernel.assign(self.conv.kernel * self.mask)
+        return self.conv(inputs)
+
+
+# Next we build our residual block layer,
+# this is just a normal residual block but with the PixelConvLayer built in
+class ResidualBlock(keras.layers.Layer):
+    def __init__(self, filters, **kwargs):
+        super(ResidualBlock, self).__init__(**kwargs)
+        self.a = keras.layers.ReLU()
+        self.b = keras.layers.Conv2D(filters=filters, kernel_size=1, activation="relu")
+        self.c = PixelConvLayer(
+            mask_type="B",
+            filters=filters // 2,
+            kernel_size=3,
+            activation="relu",
+            padding="same",
+        )
+        self.d = keras.layers.Conv2D(filters=filters, kernel_size=1, activation="relu")
+
+    def call(self, inputs):
+        x = self.a(inputs)
+        x = self.b(x)
+        x = self.c(x)
+        x = self.d(x)
+        return keras.layers.add([inputs, x])
+
+
+"""
+## Build the model based on the original paper
+"""
+
+inputs = keras.Input(shape=input_shape)
+x = PixelConvLayer(
+    mask_type="A", filters=128, kernel_size=7, padding="same", activation="relu"
+)(inputs)
+
+for _ in range(n_residual_blocks):
+    x = ResidualBlock(filters=128)(x)
+x = layers.ReLU()(x)
+
+for _ in range(2):
+    x = PixelConvLayer(
+        mask_type="B",
+        filters=128,
+        kernel_size=1,
+        strides=1,
+        activation="relu",
+        padding="valid",
+    )(x)
+
+out = keras.layers.Conv2D(
+    filters=1, kernel_size=1, strides=1, activation="sigmoid", padding="valid"
+)(x)
+
+PixelCNN = keras.Model(inputs, out)
```

Comment:
Instance names should be snake_case; capitalized names are for classes. So for instance:

```
class PixelCNN(Model): ...
```

In this case:

```
pixel_cnn = keras.Model(inputs, out)
```